app/controllers/submissions_controller.rb

```diff
@@ -170,19 +178,25 @@ def populate_file_manager
   end
 
   def manually_collect_and_begin_grading
+    begin
```

Comment:
It seems like we should have `:manually_collect_and_begin_grading` in the `before_action` you added (line 44). If you're worried about the flash message not appearing, that's something we can solve separately for all the `authorize!` filters, so it's better for now to treat them all the same.